csrc/scheduler/normalization_utils.cpp

```diff
@@ -827,8 +860,17 @@ bool projectBufferToInputs(
     return false;
   }
 
+  // must project to inputs otherwise don't have enough register or shared
+  // memory to store the buffers
+  int64_t max_available_buffer = getMaxRegOrSharedMemorySizeForPersistentBuffer(
+      runtime_info, persistent_buffer_info.persistent_buffers);
+  if (max_available_buffer <
+      persistent_buffer_size_info.persistent_buffer_size) {
```

Comment:
QQ: do we actually have a failing example when skipping projection causes launch failure?

Reply:
we don't have that example in the tests. I noticed that in the python benchmark I did. It depends on size of shared memory on GPU.

Reply:
I checked `pytest -v test_dropout_layernorm_fwd.py` locally, all passed. without new checking (`must project if don't have enough reg/smem`) added in this PR, this script fails.

Reply:
What gpu do you have locally? I'm just curious how this slipped through last time in the CI...